src/Lachain.Core/RPC/HTTP/HttpService.cs

```diff
@@ -179,43 +194,70 @@ private bool _Handle(HttpListenerContext context)
-            if(context.Request.IsLocal) return true;
-
-            if(_privateMethods.Contains(body["method"]!.ToString()))
+            if (args is null)
+                return "";
+            
+            string serializedParams = string.Empty;
+            
+            foreach(var param in args)
             {
-                if(string.IsNullOrEmpty(signature)) return false;
-                if(string.IsNullOrEmpty(timestamp)) return false;
-                // If unix timestamp diff is longer than 30 minutes, we dont handle it
-                if(!long.TryParse(timestamp.Trim(), out long unixTimestamp)) return false;
-                TimeSpan timeSpan = DateTimeOffset.Now.Subtract(DateTimeOffset.FromUnixTimeSeconds(unixTimestamp));
-                if(timeSpan.TotalMinutes >= 30)
-                    return false;
+                serializedParams += param.Key;
+                if (param.Value is null)
+                    continue;
+                serializedParams += (param.Value.HasValues ? SerializeParams((JObject)param.Value) : param.Value.ToString());
+            }
 
-                // Serialize params (format: key1value1key2value2...)
-                string serializedParams = string.Empty;
-                if(body["params"] != null && !string.IsNullOrEmpty(body["params"]!.ToString()))
+            return serializedParams;
+        }
+
+        private bool _CheckAuth(JObject body, HttpListenerContext context, string signature, string timestamp)
+        {
+            try
+            {
+                if(_privateMethods.Contains(body["method"]!.ToString()))
                 {
-                    var paramsObject = (JObject)body["params"]!;
-                    if(paramsObject != null)
-                    {
-                        foreach(var param in paramsObject)
-                        {
-                            serializedParams += param.Key + param.Value?.ToString();
-                        }
-                    }
-                }
+                    if(string.IsNullOrEmpty(signature)) return false;
+                    if(string.IsNullOrEmpty(timestamp)) return false;
+                    // If unix timestamp diff is longer than 30 minutes, we dont handle it
+                    if(!long.TryParse(timestamp.Trim(), out long unixTimestamp)) return false;
+                    TimeSpan timeSpan = DateTimeOffset.Now.Subtract(DateTimeOffset.FromUnixTimeSeconds(unixTimestamp));
+                    if(timeSpan.TotalMinutes >= 30)
+                        return false;
 
-                var messageToSign = body["method"]!.ToString() + serializedParams + timestamp;
-                var messageBytes = Encoding.UTF8.GetBytes(messageToSign);
-                var messageHash = System.Security.Cryptography.SHA256.Create().ComputeHash(messageBytes);
+                    // Serialize params (format: key1value1key2value2...)
+                    string serializedParams = SerializeParams(body["params"] as JObject);
```

Comment:
sometime 'params' can be a list (made of JToken or JObject) instead of JObject. Will SerializeParams work here? casting body["params"] as JObject if it is a list?

Reply:
For example "la_sendRawTransactionBatch" has list of strings in params

Reply:
Fixed